crowbar_framework/app/models/ceph_service.rb

```diff
@@ -122,13 +122,13 @@ def create_proposal
     if osd_nodes.size < 2
       osd_nodes_controller = select_nodes_for_role(nodes, "ceph-osd", "controller")
       osd_nodes_all = select_nodes_for_role(nodes, "ceph-osd")
-      # avoid controllers if possible (will likely be used for mon)
+      # avoid controllers if possible (ceph should not be used with openstack roles)
       osd_nodes_no_controller = osd_nodes_all - osd_nodes_controller
       osd_nodes = [osd_nodes, osd_nodes_no_controller, osd_nodes_all].flatten.uniq{ |n| n.name }
       osd_nodes = osd_nodes.take(2)
     end
 
-    mon_nodes = select_nodes_for_role(nodes, "ceph-mon", "controller")
+    mon_nodes = select_nodes_for_role(nodes, "ceph-mon", "ceph")
```

Comment:
Is "ceph" a valid node role? It's not selectable in the UI. Do you mean "storage"?